src/SixLabors.Fonts/Unicode/Dfa/StateMachine.cs

```diff
@@ -112,7 +112,8 @@ public void Apply(int[] input, Dictionary<string, Action<int, int, int[]>> actio
                 {
                     if (actions.TryGetValue(tag, out Action<int, int, int[]>? action))
                     {
-                        action(match.StartIndex, match.EndIndex + 1 - match.StartIndex, input);
+                        // can be optimized... range could not be used because of the oooold targets.
+                        action(match.StartIndex, match.EndIndex, input.Skip(match.StartIndex).Take(match.EndIndex + 1 - match.StartIndex).ToArray());
```

Comment:
Yeah. Annoying eh? I was thinking we might be able to use a span inside the caller action and avoid the allocation here